### PR TITLE
feat: Update references to new name 'TraceGateway' except for Kubernetes resources

### PR DESCRIPTION
**Pull request by skhalash**
## Description

Since we're favoring the new name "TraceGateway" over "TraceCollector," this PR focuses on updating all references accordingly. The deployment and related resources still maintain the name `trace-collector`. Achieving this shift involves more than just renaming strings. It requires implementing migration logic in the controller.

Changes proposed in this pull request:

- Rename the variables/comments/test cases
- Rename the Grafana dashboard
- Rename the status condition reasons and messages (for TracePipeline and Telemetry CRs)

## Related Issues and Documents

Related issues:
https://github.com/kyma-project/kyma/issues/17158

## Traceability

- [x] The PR is linked to a GitHub Issue.
- [x] New features have a milestone label set.
- [x] New features have defined acceptance criteria in a corresponding GitHub Issue, and all criteria are satisfied with this PR.
- [x] The corresponding GitHub Issue has a respective `area` label.
- [ ] The follow-up issues (if any) are linked in the `Related Issues` section.

## Testability

The feature is unit-tested:

- [x] Yes.
- [ ] No, because unit tests are not needed.
- [ ] No, because of ...

The feature is e2e-tested:

- [x] Yes.
- [ ] No, because e2e-tests are not needed.
- [ ] No, because of ...

Tests conducted for the PR:

## Codebase

- [x] My code follows the [Effective Go](https://go.dev/doc/effective_go) style guidelines.
- [x] The code was planned and designed following the defined architecture and the separation of concerns.
- [x] The code has sufficient comments, particularly for all hard-to-understand areas.
- [x] This PR adds value and shows no feature creep.
- [x] I have augmented the test suite that proves my fix is effective or that my feature works.
- [x] Adjusted the documentation if the change is user-facing.

### CODE REVIEW

File: controllers/telemetry/tracepipeline_controller_test.go [hunk 1]
```diff
@@ -221,7 +221,7 @@ var _ = Describe("Deploying a TracePipeline", Ordered, func() {
 		}, timeout, interval).Should(Succeed())
 	})
 
-	It("updates Trace Collector Secret when referenced secret changes", func() {
+	It("updates Trace Gateway Secret when referenced secret changes", func() {
```
Comment (NHingerl) [inline controllers/telemetry/tracepipeline_controller_test.go:224]:
```suggestion
	It("updates trace gateway Secret when referenced Secret changes", func() {
```

Gateway = lower case according to https://opentelemetry.io/docs/collector/deployment/gateway/

Is the 2nd occurrence of "secret" not a Kubernetes resource (which are upper case acc to [our guidelines](https://github.com/kyma-project/community/blob/main/docs/guidelines/content-guidelines/02-style-and-terminology.md#capitalization)) ?

File: internal/reconciler/tracepipeline/status_test.go [hunk 2]
```diff
@@ -24,7 +24,7 @@ func TestUpdateStatus(t *testing.T) {
 	_ = clientgoscheme.AddToScheme(scheme)
 	_ = telemetryv1alpha1.AddToScheme(scheme)
 
-	t.Run("should add pending condition if trace collector deployment is not ready", func(t *testing.T) {
+	t.Run("should add pending condition if trace gateway deployment is not ready", func(t *testing.T) {
```
Comment (NHingerl) [inline internal/reconciler/tracepipeline/status_test.go:27]:
```suggestion
	t.Run("should add pending condition if trace gateway Deployment is not ready", func(t *testing.T) {
```

File: internal/reconciler/tracepipeline/status_test.go [hunk 3]
```diff
@@ -57,7 +57,7 @@ func TestUpdateStatus(t *testing.T) {
 		require.Equal(t, updatedPipeline.Status.Conditions[0].Reason, reconciler.ReasonTraceGatewayDeploymentNotReady)
 	})
 
-	t.Run("should add running condition if trace collector deployment is ready", func(t *testing.T) {
+	t.Run("should add running condition if trace gateway deployment is ready", func(t *testing.T) {
```
Comment (NHingerl) [inline internal/reconciler/tracepipeline/status_test.go:60]:
```suggestion
	t.Run("should add running condition if trace gateway Deployment is ready", func(t *testing.T) {
```

File: internal/reconciler/tracepipeline/status_test.go [hunk 4]
```diff
@@ -90,7 +90,7 @@ func TestUpdateStatus(t *testing.T) {
 		require.Equal(t, updatedPipeline.Status.Conditions[0].Reason, reconciler.ReasonTraceGatewayDeploymentReady)
 	})
 
-	t.Run("should reset conditions and add pending condition if trace collector deployment becomes not ready again", func(t *testing.T) {
+	t.Run("should reset conditions and add pending condition if trace gateway deployment becomes not ready again", func(t *testing.T) {
```
Comment (NHingerl) [inline internal/reconciler/tracepipeline/status_test.go:93]:
```suggestion
	t.Run("should reset conditions and add pending condition if trace gateway Deployment becomes not ready again", func(t *testing.T) {
```

File: internal/reconciler/tracepipeline/status_test.go [hunk 5]
```diff
@@ -177,7 +177,7 @@ func TestUpdateStatus(t *testing.T) {
 		require.Equal(t, updatedPipeline.Status.Conditions[0].Reason, reconciler.ReasonReferencedSecretMissing)
 	})
 
-	t.Run("should add running condition if referenced secret exists and trace collector deployment is ready", func(t *testing.T) {
+	t.Run("should add running condition if referenced secret exists and trace gateway deployment is ready", func(t *testing.T) {
```
Comment (NHingerl) [inline internal/reconciler/tracepipeline/status_test.go:180]:
```suggestion
	t.Run("should add running condition if referenced Secret exists and trace gateway Deployment is ready", func(t *testing.T) {
```

**Review by NHingerl — APPROVED** [review]

**Review by chrkl — APPROVED** [review]

**Pull request merged**